netket/experimental/operator/_fermions_2nd.py

```diff
@@ -0,0 +1,605 @@
+            x,
+            sections,
+            self.max_conn_size,
+            self._orb_idxs,
+            self._daggers,
+            self._weights,
+            self._term_ends,
+            self._constant,
+            pad,
+        )
+
+    @staticmethod
+    @jit(nopython=True)
+    def _flattened_kernel(
+        x,
+        sections,
+        max_conn,
+        orb_idxs,
+        daggers,
+        weights,
+        term_ends,
+        constant,
+        pad=False,
+    ):
+        x_prime = np.empty((x.shape[0] * max_conn, x.shape[1]), dtype=x.dtype)
+        mels = np.zeros((x.shape[0] * max_conn), dtype=weights.dtype)
+
+        def is_empty(site):
+            return site == 0.0
+
+        def flip(site):
+            return 1 - site
+
+        # loop over the batch dimension
+        n_c = 0
+        for b in range(x.shape[0]):
+
+            xb = x[b, :]
+
+            # we can already fill up with default values
+            if pad:
+                x_prime[b * max_conn : (b + 1) * max_conn, :] = np.copy(xb)
+
+            if constant != 0.0:
+                x_prime[n_c, :] = np.copy(xb)
+                mels[n_c] = constant
+                n_c += 1
+
+            new_term = True
+            # loop over all terms and sum where necessary
+            for orb_idx, dagger, weight, term_end in zip(
+                orb_idxs, daggers, weights, term_ends
+            ):
+                if new_term:
+                    new_term = False
+                    xt = np.copy(xb)
+                    mel = weight
+                    has_xp = True
+
+                if has_xp:  # previous term items made a zero, so skip
+
+                    empty_site = is_empty(xt[orb_idx])
+
+                    if dagger:
+                        if not empty_site:
+                            has_xp = False
+                        else:
+                            mel *= (-1) ** np.sum(xt[:orb_idx])  # jordan wigner sign
+                            xt[orb_idx] = flip(xt[orb_idx])
+                    else:
+                        if empty_site:
+                            has_xp = False
+                        else:
+                            mel *= (-1) ** np.sum(xt[:orb_idx])  # jordan wigner sign
+                            xt[orb_idx] = flip(xt[orb_idx])
+
+                # if this is the end of the term, we collect things
+                if term_end:
+
+                    if has_xp:
+                        x_prime[n_c, :] = xt
+                        mels[n_c] = mel
+                        n_c += 1
+
+                    new_term = True
+
+            if pad:
+                n_c = (b + 1) * max_conn
+
+            sections[b] = n_c
+
+        if pad:
+            return x_prime, mels
+        else:
+            return x_prime[:n_c], mels[:n_c]
+
+    @staticmethod
+    def identity(hilbert):
+        """identity operator"""
+        return FermionOperator2nd(hilbert, [], [], constant=1.0)
```

Comment:
can we put this in `nkx.operator.fermion.identity`?  also the `zero`.

Reply:
I'd rather not, this is very specific to the Operator. Also, for PauliStrings it's part of the Operator.